lib/web_ui/dev/safari.dart

```diff
@@ -23,37 +23,50 @@ class Safari extends Browser {
   /// [Uri] or a [String].
   factory Safari(Uri url, {bool debug = false}) {
     version = SafariArgParser.instance.version;
-
+    mobileBrowser = SafariArgParser.instance.mobileBrowser;
     assert(version != null);
     return Safari._(() async {
-      // TODO(nurhan): Configure info log for LUCI.
-      final BrowserInstallation installation = await getOrInstallSafari(
-        version,
-        infoLog: DevNull(),
-      );
-
-      // In the latest versions of MacOs opening Safari browser with a file brings
-      // a popup which halts the test.
-      // The following list of arguments needs to be provided to the `open` command
-      // to open Safari for a given URL. In summary they provide a new instance
-      // to open, that instance to wait for opening the url until Safari launches,
-      // provide Safari bundles identifier.
-      // The details copied from `man open` on MacOS.
-      // TODO(nurhan): https://github.com/flutter/flutter/issues/50809
-      var process = await Process.start(installation.executable, [
-        '-F', // Open a fresh application with no persistant state.
-        '-W', // Open to wait until the applications it opens.
-        '-n', // Open a new instance of the application.
-        '-b', // Specifies the bundle identifier for the application to use.
-        'com.apple.Safari', // Bundle identifier for Safari.
-        '${url.toString()}'
-      ]);
-
-      return process;
+      if (mobileBrowser) { // iOS-Safari
+        var process = await Process.start('xcrun', [
+          'simctl', // Open a fresh application with no persistant state.
+          'openurl', // Open to wait until the applications it opens.
+          'booted', // Open a new instance of the application.
+          '${url.toString()}'
+        ]);
+
+        return process;
+      } else { // Desktop-Safari
+        // TODO(nurhan): Configure info log for LUCI.
+        final BrowserInstallation installation = await getOrInstallSafari(
+          version,
+          infoLog: DevNull(),
+        );
+
+        // In the latest versions of MacOs opening Safari browser with a file brings
+        // a popup which halts the test.
+        // The following list of arguments needs to be provided to the `open` command
+        // to open Safari for a given URL. In summary they provide a new instance
+        // to open, that instance to wait for opening the url until Safari launches,
+        // provide Safari bundles identifier.
+        // The details copied from `man open` on MacOS.
+        // TODO(nurhan): https://github.com/flutter/flutter/issues/50809
+        var process = await Process.start(installation.executable, [
+          '-F', // Open a fresh application with no persistant state.
+          '-W', // Open to wait until the applications it opens.
+          '-n', // Open a new instance of the application.
+          '-b', // Specifies the bundle identifier for the application to use.
```

Comment:
Here and above please replace "application" and "applications" with "Safari". It is reasonable for the docs for `open` to say "application" because those doc don't know which app you are opening. However, we are specifically opening a single instance of Safari.

Reply:
done.